bugbug/repository.py

```diff
@@ -310,27 +310,57 @@ def _hg_log(revs):
     return hg_log(HG, revs)
 
 
-def get_revs(hg, date_from=None):
-    # TODO: Retrieve all commits once calculate_experiences is faster and less memory hungry.
-    if date_from is not None:
-        pushdate = (date_from - relativedelta(months=12)).strftime("%Y-%m-%d")
-        rev_start = f"pushdate('>{pushdate}')"
-    else:
-        rev_start = 0
-
-    print(f"Getting revs from {rev_start} to tip...")
+def get_revs(hg):
+    print(f"Getting revs from 0 to tip...")
 
     args = hglib.util.cmdbuilder(
-        b"log",
-        template="{node}\n",
-        no_merges=True,
-        branch="central",
-        rev=f"{rev_start}:tip",
+        b"log", template="{node}\n", no_merges=True, branch="central", rev=f"0:tip"
     )
     x = hg.rawcommand(args)
     return x.splitlines()
 
 
+class exp_queue:
+    def __init__(self, start_day, maxlen, default):
+        self.list = deque([default] * maxlen, maxlen=maxlen)
```

Comment:
when init with default=[] then it's the same instance of default everywhere in queue.
So for example q[0].append(1) will give deque([1], [1], ...).
Is it expected ?

Reply:
Elements of the queue are never modified. I've used a tuple now, so this is basically asserted.